src/sinks/util/encoding.rs

```diff
@@ -0,0 +1,504 @@
+//! This module holds encoding related code.
+//!
+//! You'll find two stuctures for configuration:
+//!   * `EncodingConfig<E>`: For sinks without a default `Encoding`.
+//!   * `EncodingConfigWithDefault<E: Default>`: For sinks that have a default `Encoding`.
+//!
+//! Your sink should define some `Encoding` enum that is used as the `E` parameter.
+//!
+//! You can use either of these for a sink! They both implement `EncodingConfiguration`, which you
+//! will need to import as well.
+//!
+//! # Using a configuration
+//!
+//! To use an `EncodingConfig` involves three steps:
+//!
+//!  1. Use `#[serde(deserialize_with = "EncodingConfig::from_deserializer")]` or
```

Comment:
we should make sure this is updated to the new change we talked about.

Reply:
Done